test/index.test.js

```diff
@@ -1,41 +1,218 @@
 'use strict';
 
-const expect = require('chai').expect;
-const Hapi   = require('hapi');
+const Bluebird = require('bluebird');
+const expect   = require('chai').expect;
+const Hapi     = require('hapi');
+const Redis    = require('then-redis');
+
+const redisClient = Redis.createClient({
+  port: '6379',
+  host: 'localhost',
+  enable_offline_queue: false
+});
 
 describe('plugin', () => {
 
+  const shortLimitRate = { limit: 1, window: 60 };
+  const shortWindowRate = { limit: 10, window: 1 };
+  const defaultRate = { limit: 10, window: 60 };
+
+  const overLimitErrMessage = 'over limit';
+
   const server = new Hapi.Server();
+
   server.connection({ port: 80 });
+  server.route([{
+    method: 'POST',
+    path: '/defaultTest',
+    config: {
+      handler: (request, reply) => {
+        reply({ rate: request.plugins['hapi-rate-limit'].rate });
+      }
+    }
+  },
+  {
+    method: 'POST',
+    path: '/shortLimitTest',
+    config: {
+      plugins: {
+        rateLimit: shortLimitRate
+      },
+      handler: (request, reply) => {
+        reply({ rate: request.plugins['hapi-rate-limit'].rate });
+      }
+    }
+  },
+  {
+    method: 'POST',
+    path: '/shortWindowTest',
+    config: {
+      plugins: {
+        rateLimit: shortWindowRate
+      },
+      handler: (request, reply) => {
+        reply({ rate: request.plugins['hapi-rate-limit'].rate });
+      }
+    }
+  },
+  {
+    method: 'POST',
+    path: '/disabledTest',
+    config: {
+      plugins: {
+        rateLimit: {
+          enabled: false
+        }
+      },
+      handler: (request, reply) => {
+        reply({ rate: request.plugins['hapi-rate-limit'].rate });
+      }
+    }
+  },
+  {
+    method: 'PUT',
+    path: '/putTest',
+    config: {
+      handler: (request, reply) => {
+        reply({ rate: request.plugins['hapi-rate-limit'].rate });
+      }
+    }
+  }]);
 
-  server.register([
-    require('../lib')
-  ], (err) => {
-    if (err) {
-      throw err;
+  server.register([{
+    register: require('../lib'),
+    options: {
+      defaultRate,
+      redisClient,
+      overLimitError: new Error(overLimitErrMessage)
```

Comment:
instead of creating a new error here, you should pass in an error constructor that your plugin can [create new instances of](https://github.com/lob/lob-api/blob/master/src/plugins/services/fetchOrCreate.js#L50).